plotly/tools.py

```diff
@@ -1477,6 +1477,45 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def _make_colorscale(colors, scale=None):
+        """
+        Makes a colorscale from a list of colors and scale
+
+        Takes a list of colors and scales and constructs a colorscale based
+        on the colors in sequential order. If 'scale' is left empty, a linear-
+        interpolated colorscale will be generated. If 'scale' is a specificed
+        list, it must be the same legnth as colors and must contain all floats
+        For documentation regarding to the form of the output, see
+        https://plot.ly/python/reference/#mesh3d-colorscale
+        """
+        colorscale = []
+
+        if not scale:
+            for j in range(len(colors)):
```

Comment:
🐄 `for i, color in enumerate(colors):`

No big deal, it just gets around the `colors[i]` below.